lib/Data/Validate/Sanctions/Redis.pm

```diff
@@ -0,0 +1,230 @@
+    }
+}
+
+sub set_sanction_file {
+    die 'Not applicable';
+}
+
+sub get_sanction_file {
+    die 'Not applicable';
+}
+
+sub get_sanctioned_info {
+    my $self = shift;
+
+    die "This function can only be called on an object" unless $self;
+
+    return Data::Validate::Sanctions::get_sanctioned_info($self, @_);
+}
+
+sub _load_data {
+    my $self = shift;
+
+    $self->{last_time}               //= 0;
+    $self->{_data}                   //= {};
+    $self->{_sanctioned_name_tokens} //= {};
+    $self->{_token_sanctioned_names} //= {};
+
+    my $last_time = $self->{last_time};
+    for my $source ($self->{sources}->@*) {
+        try {
+            my $updated = $self->{redis_read}->hget("SANCTIONS::$source", 'updated') // 0;
+            next if $updated <= ($self->{_data}->{$source}->{updated} // 0);
+
+            $self->{_data}->{$source}->{content}  = decode_json_utf8($self->{redis_read}->hget("SANCTIONS::$source", 'content'));
+            $self->{_data}->{$source}->{verified} = $self->{redis_read}->hget("SANCTIONS::$source", 'verified');
+            $self->{_data}->{$source}->{updated}  = $updated;
+            $last_time                            = $updated if $updated > $last_time;
+        } catch {
+            $self->{_data}->{$source}->{content}  = [];
+            $self->{_data}->{$source}->{updated}  = 0;
+            $self->{_data}->{$source}->{verified} = 0;
+            $self->{_data}->{$source}->{error}    = "Failed to load from Redis: $@";
+        }
+    }
+    $self->{last_time} = $last_time;
+
+    $self->_index_data();
+
+    foreach my $sanctioned_name (keys $self->{_index}->%*) {
+        my @tokens = Data::Validate::Sanctions::_clean_names($sanctioned_name);
+        $self->{_sanctioned_name_tokens}->{$sanctioned_name} = \@tokens;
+        foreach my $token (@tokens) {
+            $self->{_token_sanctioned_names}->{$token}->{$sanctioned_name} = 1;
+        }
+    }
+
+    return $self->{_data};
+}
+
+sub _save_data {
+    my $self = shift;
+
+    die 'Redis write connection is missing' unless $self->{redis_write};
+
+    for my $source ($self->{sources}->@*) {
+        $self->{_data}->{$source}->{verified} = time;
+        $self->{redis_write}->hmset(
+            "SANCTIONS::$source",
+            'updated'  => $self->{_data}->{$source}->{updated} // 0,
+            'content'  => encode_json_utf8($self->{_data}->{$source}->{content} // []),
+            'verified' => $self->{_data}->{$source}->{verified},
+            'error'    => $self->{_data}->{$source}->{error} // ''
+        );
+    }
+
+    return;
+}
+
+sub _default_sanction_file {
+    die 'Not applicable';
+}
+
+sub data {
+    my ($self) = @_;
+
+    return $self->{_data};
+}
+
+1;
+__END__
+
+=encoding utf-8
+
+=head1 NAME
+
+Data::Validate::Sanctions::Redis - An extention of L<Data::Validate::Sanctions> that stores sanction data in redis rather than a local file.
+
+=head1 SYNOPSIS
+    # it only works with OO calls
```

Comment:
What does this mean?

Reply:
I removed it. I wanted to tell that unlike the parent class `Data::Validate::Sanctions`, this one cannot be used as a procedural module. I removed it, because it's self-evident.